### ISSUE TITLE
Feat: Support for Object Parameters in Tekton Pipelines

### ISSUE THREAD

**Issue by luismafra** [IBM/tekton-lint#116]
**Is your feature request related to a problem? Please describe.**

Currently, the `tekton-lint` tool does not properly handle parameters of type `object` in Tekton pipeline definitions. This causes issues when parameters of this type are used in tasks, resulting in incorrect warnings and errors.

For example, given a parameter defined as:

```yaml
- name: gitRepository
  description: Git Repository.
  type: object
  properties:
    defaultBranch:
      type: string
    name:
      type: string
    owner:
      type: string
```
And its usage in a task:

```yaml
- name: repositoryName
  value: $(params.gitRepository.name)
 ```
The tool incorrectly reports:

```bash
Undefined param 'gitRepository.name' at .spec.tasks[2].params[8].value in 'git-clone'
```

**Describe the solution you'd like**
The tool should correctly recognize and handle parameters of type object and their properties, without reporting them as undefined.

**Describe alternatives you've considered**

**Additional context**
https://tekton.dev/docs/pipelines/tasks/#object-type
